cmake/Modules/FindComputeCpp.cmake

```diff
@@ -370,23 +414,48 @@ endfunction()
 #  binaryDir : Intermediate directory to output the integration header.
 #  sourceFiles : Source files to be compiled for SYCL.
 #
-function(add_sycl_to_target targetName binaryDir sourceFiles)
+function(add_sycl_to_target)
+  set(options)
+  set(oneValueArgs TARGET BINARY_DIR)
+  set(multiValueArgs SOURCES)
+  cmake_parse_arguments(add_sycl_to_target "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  set(targetName ${add_sycl_to_target_TARGET})
+  set(binaryDir ${add_sycl_to_target_BINARY_DIR})
+  set(sourceFiles ${add_sycl_to_target_SOURCES})
 
-  set(sourceFiles ${sourceFiles} ${ARGN})
   set(fileCounter 0)
   target_include_directories(
     ${targetName} SYSTEM
     PRIVATE ${OpenCL_INCLUDE_DIR}
     PRIVATE ${COMPUTECPP_INCLUDE_DIRECTORY}
   )
-  # Add custom target to run compute++ and generate the integration header
-  foreach(sourceFile ${sourceFiles})
-    if(NOT IS_ABSOLUTE ${sourceFile})
```

Comment:
I think that you have removed this check to ensure that the source files have absolute paths. The custom compute++ command requires that all source files have absolute not relative paths.